gpii/node_modules/singleInstance/test/SingleInstanceTests.js

```diff
@@ -0,0 +1,164 @@
+/*
+ * singleInstance Tests
+ *
+ * Copyright 2017 Raising the Floor - International
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * The R&D leading to these results received funding from the
+ * Department of Education - Grant H421A150005 (GPII-APCP). However,
+ * these results do not necessarily represent the policy of the
+ * Department of Education, and you should not assume endorsement by the
+ * Federal Government.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+"use strict";
+
+var fluid = require("universal"),
+    fs = require("fs"),
+    os = require("os"),
+    child_process = require("child_process");
+
+var jqUnit = fluid.require("node-jqunit");
+var gpii = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.tests.singleInstance");
+
+require("singleInstance");
+
+jqUnit.module("gpii.tests.singleInstance");
+
+jqUnit.test("Testing checkInstance, registerGpiiInstance, and deregisterInstance", function () {
+
+    var pidFile = os.tmpdir() + "/gpii-test-pid" + Date.now();
+
+    // Start another process (a cross-platform "sleep"), to test against a live pid.
+    var otherProcess = child_process.exec("node -e \"setTimeout(console.log, 5000)\"");
+
+    var tests = [
```

Comment:
The most direct thing will probably be to just create a little ModelComponent http://docs.fluidproject.org/infusion/development/tutorial-gettingStartedWithInfusion/ModelComponents.html to hold the mutable state, use its applier to set process.pid, otherProcess.pid etc. when they occur and then write a little string dialect in the definitions to indirect into it - e.g. file: "process.pid", checkInstance: "otherProcess.pid", and then use fluid.get(myComponent, expected.file) etc. to fish them out again


Your other option is to do what the LifecycleManager does and just throw the entire test configuration into fluid.expand with a suitable "fetcher" which will then resolve references of the form "${process}.pid" but the comment above this line suggests that the time might not be right for this - https://github.com/GPII/universal/blob/master/gpii/node_modules/lifecycleManager/src/Resolvers.js#L80